src/components/PopupWithForm.js

```diff
@@ -1,28 +1,35 @@
 import Popup from "./Popup.js";
-
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
-    this._inputList = this._popupForm.querySelectorAll(".modal__input");
+    this._submitBtn = this._popupElement.querySelector(".modal__save-button");
     this._handleFormSubmit = handleFormSubmit;
-    this.setEventListeners();
   }
 
-  getForm() {
-    return this._popupForm;
+  close() {
+    this._popupForm.reset();
```

Comment:
You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` or when you close the modal, because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.